lib/engine/step/g_1822/destination_token.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+
+module Engine
+  module Step
+    module G1822
+      class DestinationToken < Base
+        ACTIONS = %w[hex_token].freeze
+
+        def actions(entity)
+          return [] unless entity == current_entity
+          return [] unless can_place_token?(entity)
+
+          ACTIONS
+        end
+
+        def available_tokens(entity)
+          destination_token = entity.find_token_by_type(:destination)
+          return [] unless destination_token
+
+          [destination_token]
+        end
+
+        def can_place_token?(entity)
+          return false if !entity.corporation? || (entity.corporation? && entity.type != :major)
+          return false if available_tokens(entity).empty?
+
+          destination_hex = @game.class::DESTINATIONS[entity.id]
+          parts = @game.graph.connected_nodes(entity).keys
```

Comment:
you cannot call this unless you hide it under a @game.loading check or you use auto actions

it's too expensive